src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -254,11 +260,52 @@ public void shouldReceiveValidResponseWhenBannerFormatIsNull() throws IOExceptio
                 .isEqualTo(expectedCommonMessage);
         assertThat(capturedCommonMessage.getGreenbidsId()).isNotNull();
         assertThat(capturedCommonMessage.getBillingId()).isNotNull();
+    }
+
+    @Test
+    public void shouldReturnValidHeadersAndTimeouts() {
+        final Banner banner = givenBanner();
+
+        final ObjectNode impExtNode = mapper.createObjectNode();
+        impExtNode.set("gpid", TextNode.valueOf("gpidvalue"));
+        impExtNode.set("prebid", givenPrebidBidderParamsNode());
+
+        final Imp imp = Imp.builder()
+                .id("adunitcodevalue")
+                .ext(impExtNode)
+                .banner(banner)
+                .build();
+
+        final AuctionContext auctionContext = givenAuctionContext(context -> context, List.of(imp), true);
+        final AuctionEvent event = AuctionEvent.builder()
+                .auctionContext(auctionContext)
+                .bidResponse(auctionContext.getBidResponse())
+                .build();
+
+        final HttpClientResponse mockResponse = mock(HttpClientResponse.class);
+        when(mockResponse.getStatusCode()).thenReturn(202);
+        when(httpClient.post(anyString(), any(MultiMap.class), anyString(), anyLong()))
+                .thenReturn(Future.succeededFuture(mockResponse));
+
+        // when
+        final Future<Void> result = target.processEvent(event);
+
+        // then
+        verify(httpClient).post(
+                urlCaptor.capture(),
+                headersCaptor.capture(),
+                jsonCaptor.capture(),
+                timeoutCaptor.capture());
```

Comment:
it's not necessary to capture everything, I believe for the url and timeout you can use an argument matcher, and for the json just any matcher (since it's not checked in this use case)

please check other tests as well, don't need to capture when it's much easier to use existing argument matchers